plugins/tableselection/plugin.js

```diff
@@ -508,7 +509,13 @@
 			}
 		}
 
-		tmpContainer.setHtml( evt.data.dataValue );
+		if ( !dataProcessor ) {
+			dataProcessor = new CKEDITOR.htmlDataProcessor( editor );
+		}
+
+		// Pasted value must be filtered using dataProcessor to strip all unsafe code
+		// before inserting it into temporary container.
+		tmpContainer.setHtml( dataProcessor.toHtml( evt.data.dataValue ) );
```

Comment:
Use options argument with `fixForBody` so that no extra blocks are added, like so:
```
		tmpContainer.setHtml( dataProcessor.toHtml( evt.data.dataValue, {
			fixForBody: false
		} ) );
```